backend/core/views.py

```diff
@@ -144,6 +149,20 @@ def summary_stats(self, request, pk=None):
     # def shuffle_reviewers(self, request, pk=None):
     #     return Response("TODO")
 
+    @action(
+        detail=False,
+        methods=["POST"],
+        serializer_class=CardSummarySerializer,
+        permission_classes=[HasObjectPermission(permissions=Team.PERMISSION_MANAGE_CARDS)]
+    )
+    def bulk_set_due_dates(self, request, pk=None):
+
+        team: models.Team = models.Team.objects.filter(name=request.data.get('team')).first()
+        team_cards = get_team_cards(team, request.data.get('content_item'))
+        [card.set_due_time(request.data.get('due_time')) for card in team_cards]
+
+        return Response(AgileCardSerializer([card for card in team_cards][0]).data)
```

Comment:
I think the card summary serialiser can be used at the end. It returns less data so it's more efficient. Also, chances are that this endpoint will be accessed from the card summary page so it would make sense to make use of serialisers used for that page

Reply:
I tried using the CardSummarySerializer here but I keep on getting this error:

Got AttributeError when attempting to get a value for field `team` on serializer `CardSummarySerializer`.
The serializer field might be named incorrectly and not match any attribute or key on the `AgileCard` instance.
Original exception text was: 'AgileCard' object has no attribute 'team'

The error doesn't make sense to me since I did add an extra field 'team' to the CardSummarySerializer?

Reply:
Sorted this error out, I now make use of CardSummarySerializer only.